### PR TITLE
Make doctests IDE (PyCharm) runnable

### PR DESCRIPTION
**Pull request by HyukjinKwon**
We can run doctests via PyCharm:

Screen Shot 2019-04-26 at 1 22 51 PM

**Before:**

Screen Shot 2019-04-26 at 1 22 58 PM

Screen Shot 2019-04-26 at 1 21 10 PM

**After:**

Screen Shot 2019-04-26 at 1 26 17 PM

Screen Shot 2019-04-26 at 1 25 54 PM

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -189,7 +190,7 @@ def spark_type(self):
         return self.schema.fields[-1].dataType
 
     def astype(self, dtype):
-        from databricks.koalas.typing import as_spark_type
+        from databricks.koalas.typehint import as_spark_type
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py]: `typing` -> `typehint` was a workaround for, something I suspect a bug in PyCharm:

```
Traceback (most recent call last):
  File "/.../pycharm/docrunner.py", line 297, in <module>
    modules = [loadSource(a[0])]
  File "/.../pycharm/docrunner.py", line 229, in loadSource
    module = imp.load_source(moduleName, fileName)
  File "/.../lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/.../koalas/databricks/koalas/series.py", line 24, in <module>
    import pandas as pd
  File "/.../lib/python3.7/site-packages/pandas/__init__.py", line 23, in <module>
    from pandas.compat.numpy import *
  File "/.../lib/python3.7/site-packages/pandas/compat/__init__.py", line 446, in <module>
    import typing
  File "/.../koalas/databricks/koalas/typing.py", line 33, in <module>
    T = typing.TypeVar("T")
AttributeError: module 'typing' has no attribute 'TypeVar'
```
Looks executing doctests fails to resolve namespace propery and it looks getting confused by, current module's `typing` and Python built-in's `typing`.

File: databricks/koalas/__init__.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def assert_pyspark_version():
 from databricks.koalas.namespace import *
 from databricks.koalas.frame import DataFrame
 from databricks.koalas.series import Series
-from databricks.koalas.typing import Col, pandas_wrap
+from databricks.koalas.type import Col, pandas_wrap
```
Comment (HyukjinKwon) [inline databricks/koalas/__init__.py]: `typing` -> `type` was a workaround for, something I suspect a bug in PyCharm:

```
Traceback (most recent call last):
  File "/.../pycharm/docrunner.py", line 297, in <module>
    modules = [loadSource(a[0])]
  File "/.../pycharm/docrunner.py", line 229, in loadSource
    module = imp.load_source(moduleName, fileName)
  File "/.../lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/.../koalas/databricks/koalas/series.py", line 24, in <module>
    import pandas as pd
  File "/.../lib/python3.7/site-packages/pandas/__init__.py", line 23, in <module>
    from pandas.compat.numpy import *
  File "/.../lib/python3.7/site-packages/pandas/compat/__init__.py", line 446, in <module>
    import typing
  File "/.../koalas/databricks/koalas/typing.py", line 33, in <module>
    T = typing.TypeVar("T")
AttributeError: module 'typing' has no attribute 'TypeVar'
```
Looks executing doctests fails to resolve namespace propery and it looks getting confused by, current module's `typing` and Python built-in's `typing`.

File: databricks/koalas/series.py [hunk 3]
```diff
@@ -28,6 +28,7 @@
 from pyspark.sql.types import FloatType, DoubleType, LongType, StructType, TimestampType, \
     to_arrow_type
 
+from databricks import koalas as ks
```
Comment (rxin) [inline databricks/koalas/series.py]: Is this required?

Comment (HyukjinKwon) [reply databricks/koalas/series.py]: It's required for PyCharm IDE to run doctests (but not the library itself)

1. PyCharm's doctest running requires it.
2. PyCharm reference resolving requires it

(see before after images in PR description)

Comment (HyukjinKwon) [reply databricks/koalas/series.py]: It is not required if we run the tests via terminal.

Comment (rxin) [reply databricks/koalas/series.py]: OK I'd then leave a comment here, because otherwise people might remove it in the future.

**Comment by rxin** [comment]
LGTM other than that one thing.

File: databricks/koalas/frame.py [hunk 4]
```diff
@@ -27,6 +27,7 @@
 from pyspark.sql.types import StructType, to_arrow_type
 from pyspark.sql.utils import AnalysisException
 
+from databricks import koalas as ks
```
Comment (rxin) [inline databricks/koalas/frame.py]: add comment explaining why we are importing koalas as ks

Comment (HyukjinKwon) [reply databricks/koalas/frame.py]: yup

**Pull request merged**